### ISSUE TITLE
/dev/ttyS1 does not exist under HypriotOS

### ISSUE THREAD

**Issue by TomTom101** [angelnu/docker-ccu2#14]
I am trying to run CCU2 under HypriotOS, amongst other (fixable) things this error occurs when running build.sh:

_build.sh_ terminates w/ error
```docker: Error response from daemon: linux runtime spec devices: error gathering device information while adding custom device "/dev/ttyS1": lstat /dev/ttyS1: no such file or directory.```

`$ uname -a`
> Linux black-pearl 4.4.50-hypriotos-v7+ #1 SMP PREEMPT Sun Mar 19 14:11:54 UTC 2017 armv7l GNU/Linux

I have to add I could not yet get the HM-MOD-RPI-PCB radio to run within this container. The HW itself does work, I cannot get them to communicate.

Whatfor is /dev_org/ttyS1 used anyway?

**Comment by angelnu** [comment]
I split the build script to not require any serial port. The new deploy.sh will also only pass those devices available and not assume any particular platform.

With this the container should build and deploy in any ARM platform. Please give it a try.

**Comment by TomTom101** [comment]
Wow, quick reaction! That's commitment! Unfortunately, I can't get RFD/BidCos to run outside the container either. I did all the necessary changes in _/boot/cmdline.txt_,  _/boot/config.txt_, disabled _hciuart.service_. I am out of ideas of how I can further debug. What does bug me is that _/dev/serial0_ still links to _ttyS0_, I was expecting _ttyAMA0_ here. Not sure if this is relevant or not. Any idea what I could check? Thanks a lot!

**Comment by angelnu** [comment]
@TomTom101 my proposal would be to wait for the homematic IP USB adapter that ELV announced for their next issue. Then we do not have to worry about the PI serial port and the special device driver. I am myself using a LAN gateway as antenna so now I can start the CCU2 docker container in any of my Raspis/OrangePIs.

Each Linux distro uses a different dev file for the serial port. I am using an Orange PI and there it is /dev/ttyS0. If this would be the problem a fix in the entrypoint.sh would address it.

How did you check that the adapter is not working outside the docker? Did you try to run rfd manually on the host?

**Comment by TomTom101** [comment]
I have a Raspbian installation on a different cart where I used the RFD https://github.com/eq-3/occu to see whether the antenna is good to begin with. It does connect there. I did the same on the Hypriot distro (host) without avail. Hence I cannot expect it to work in the docker container either.

One difference I found between my Raspian install, where I can connect to the antenna and Hypriot. I cannot set GPIO18 low on the latter:
`cat /sys/class/gpio/gpio18/direction`
_out_

`cat /sys/class/gpio/gpio18/value`
_1_

```echo 0 > cat /sys/class/gpio/gpio18/value
cat /sys/class/gpio/gpio18/value```
_1_

Wouldn't that be important? I have no clue whatsoever what I could do to fix this.

You say the HM adapter is not yet available but announced in their journal? The "HmIP-RF USB"? I'd hope that it should make things easier.
